src/deployer/Deployer.ts

```diff
@@ -47,10 +48,91 @@ export class Deployer {
     return adapter.toInstance(contract, contractAddress, parameters);
   }
 
-  public async deployed<T, A = T, I = any>(
-    contract: Instance<A, I> | (T extends Truffle.Contract<I> ? T : never),
-    contractIdentifier?: string,
-  ): Promise<I> {
+  public async deployERC1967Proxy<A, I = any>(
+    implementationFactory: Instance<A, I>,
+    argsOrParameters: OverridesAndLibs | TypedArgs<A> = [] as TypedArgs<A>,
+    parameters: OverridesAndLibs = {},
+  ) {
+    return this.deployProxy(
+      implementationFactory,
+      "ERC1967Proxy",
+      (implementationAddress) => [implementationAddress, "0x"],
+      argsOrParameters,
+      parameters,
+    );
+  }
+
+  public async deployTransparentUpgradeableProxy<A, I = any>(
```

Comment:
Can we add Adminable proxy support from Solarity 3.0?

Reply:
Yep, added